File: problems_0-1000/605-CanPlaceFlowers/java/Solution.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int length = flowerbed.length;
        int i = 0;

        while (i < length && n > 0) {
            if (flowerbed[i] == 0) {
                boolean leftEmpty = (i == 0 || flowerbed[i - 1] == 0);
                boolean rightEmpty = (i == length - 1 || flowerbed[i + 1] == 0);

                if (leftEmpty && rightEmpty) {
                    flowerbed[i] = 1;
                    n--;
                    i++; // Skip the next plot, as it's now occupied
                }
            }
            i++;
        }

        return n <= 0;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        int[] flowerbed = {1, 0, 0, 0, 1};
        int n = 1;
        System.out.println(solution.canPlaceFlowers(flowerbed, n));
    }
}

File: problems_0-1000/605-CanPlaceFlowers/python/solution.py
class Solution:
    def canPlaceFlowers(self, flowerbed: list[int], n: int) -> bool:
        
        length = len(flowerbed)
        i = 0

        while i < length and n > 0:
            if flowerbed[i] == 0:
                left_empty = i == 0 or flowerbed[i-1] == 0
                right_empty = i == length - 1 or flowerbed[i+1] == 0
                if left_empty and right_empty:
                    flowerbed[i] = 1
                    n -= 1
                    i += 1 
            i += 1

        return n <= 0
    
if __name__ == '__main__':
    flowerbed = [1,0,0,0,1]
    n = 2
    s = Solution()
    print(s.canPlaceFlowers(flowerbed, n))